sim/dex-data.ts

```diff
@@ -187,6 +187,20 @@ export class RuleTable extends Map<string, string> {
 		return this.has(`*${thing}`);
 	}
 
+	isRestrictedSpecies(species: Species) {
+		if (this.has('+pokemon:' + species.id)) return false;
+		if (this.has('*pokemon:' + species.id)) return true;
+		if (this.has('+basepokemon:' + toID(species.baseSpecies))) return false;
+		if (this.has('*basepokemon:' + toID(species.baseSpecies))) return true;
+		const tier = species.tier === '(PU)' ? 'ZU' : species.tier === '(NU)' ? 'PU' : species.tier;
+		if (this.has('+pokemontag:' + toID(tier))) return false;
+		if (this.has('*pokemontag:' + toID(tier))) return true;
+		const doublesTier = species.doublesTier === '(DUU)' ? 'DNU' : species.doublesTier;
+		if (this.has('+pokemontag:' + toID(doublesTier))) return false;
+		if (this.has('*pokemontag:' + toID(doublesTier))) return true;
+		return this.has('*pokemontag:allpokemon');
```

Comment:
nitpick but shouldnt these all be template strings to remain consistent with other parts of this file?

Reply:
Oops; as I mentioned in #6967 I originally thought of putting this in a different file where the style was different.